include/tvm/runtime/container.h

```diff
@@ -1151,6 +1151,42 @@ class String : public ObjectRef {
    */
   inline String operator=(std::string other);
 
+  /*!
+   * \brief Compare is less than other std::string
+   *
+   * \param other The other string
+   *
+   * \return the comparison result
+   */
+  bool operator<(const std::string& other) const { return this->compare(other) < 0; }
```

Comment:
This was a missing part that can potentially cause mysterious bugs, because < was inherited from ObjectRef which was pointer comparison instead.